### PR TITLE
Update the homepage and add tests

### LINKED ISSUE #65: Update HomePage
Update the home page to include a button for registration and login.

### PR DESCRIPTION
**Pull request by shirazyomtov**
### The changes that were made :
1. Update the homepage.

2. Add login, sign up and redirect login to the adminUI

3. Add tests to the homepage

4. Add base_home for the homepage and future login and sign up pages

5. Add image of a lightbulb for the homepage icon
6. Add the option for users to log in via adminUI and redirect the user to the home page(temporary till we have a profile) after logging in.

Resolve #65 

image

### CODE REVIEW

**Review by shacharlevy — CHANGES_REQUESTED** [review]
Nice work! I think it is necessary to update the CSS file related to the homePage.html

**Comment by shirazyomtov** [comment]
> Nice work! I think it is necessary to update the CSS file related to the homePage.html

Done, I left the style folder for the css files that we might need in the future.

**Review by Yardengz — APPROVED** [review]
LGTM

**Review by shacharlevy — DISMISSED** [review]
LGTM

File: main/templates/main/homePage.html [hunk 1]
```diff
@@ -1,58 +1,27 @@
...
       </div>
-	</div>
-	
-</body>
-</html>
+    </div>
+  </main>
+  </body>
+{% endblock %}
+</html>
```
Comment (natifridman) [inline main/templates/main/homePage.html:27]: Missing newline at end of file

File: tests/main/test_homePage.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+@pytest.mark.django_db
+class TestHomepageView:
+    def test_user_redirected_to_admin_login(self, client):
+        response = client.get('/admin')
+        assert response.status_code == 301
+
+    def test_extends_of_base(self, client):
+        response = client.get('/')
+        assert response.status_code == 200
+        assert b'<nav' in response.content
```
Comment (natifridman) [inline tests/main/test_homePage.py:13]: Maybe we should check the template name instead? I think it should be in `response.template_name`

File: tests/main/test_homePage.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        assert b'Login' in response.content
+        assert b'Sign Up' in response.content
+
+    def test_logged_user(self, client, persist_user):
+        client.force_login(persist_user)
+        response = client.get('/')
+        assert response.status_code == 200
+        assert b'<nav' in response.content
```
Comment (natifridman) [inline tests/main/test_homePage.py:21]: same

File: tests/main/test_homePage.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import pytest
+
+
+@pytest.mark.django_db
+class TestHomepageView:
+    def test_user_redirected_to_admin_login(self, client):
```
Comment (natifridman) [inline tests/main/test_homePage.py:6]: Doesn't this test check redirect from the admin page?
```suggestion
    def test_user_redirected_from_admin_page(self, client):
```

File: tests/main/test_homePage.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        response = client.get('/admin')
+        assert response.status_code == 301
```
Comment (natifridman) [inline tests/main/test_homePage.py:8]: Maybe we can also check the response URL?

File: tests/main/test_homePage.py
Comment (natifridman) [inline tests/main/test_homePage.py:1]: We usually use [lowercase](https://peps.python.org/pep-0008/#package-and-module-names) in Python file names

**Review by natifridman — COMMENTED** [review]
Nice work! added some comments

**Review by Omeramsc — COMMENTED** [review]
Please upload photos of the webpages, it'll make it easier to understand and review

**Comment by shirazyomtov** [comment]
> Please upload photos of the webpages, it'll make it easier to understand and review

Done

File: main/templates/main/homePage.html (hunk 1 above)
Comment (shirazyomtov) [reply main/templates/main/homePage.html]: Done

File: tests/main/test_homePage.py (hunk 2 above)
Comment (shirazyomtov) [reply tests/main/test_homePage.py]: Done

File: tests/main/test_homePage.py (hunk 3 above)
Comment (shirazyomtov) [reply tests/main/test_homePage.py]: Done

File: tests/main/test_homePage.py (hunk 4 above)
Comment (shirazyomtov) [reply tests/main/test_homePage.py]: Yes, I changed it.

File: tests/main/test_homePage.py (hunk 5 above)
Comment (shirazyomtov) [reply tests/main/test_homePage.py]: Done

File: tests/main/test_homePage.py
Comment (shirazyomtov) [reply tests/main/test_homePage.py:1]: Done

**Review by shacharlevy — DISMISSED** [review]
LGTM!

File: HighTeach/middleware_login.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
+from django.shortcuts import redirect
+from django.urls import reverse
+from django.contrib.auth import authenticate
+
+
+class AdminRedirectMiddleware:
```
Comment (natifridman) [inline HighTeach/middleware_login.py:6]: Why is this here and not as [a view](https://docs.djangoproject.com/en/4.2/topics/auth/default/#limiting-access-to-logged-in-users)?

File: tests/main/test_homePage.py
Comment (natifridman) [reply tests/main/test_homePage.py:1]: It looks like it's still mixedCase

File: tests/main/test_homePage.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import pytest
+from django.urls import reverse
+
+
+@pytest.mark.django_db
+class TestHomepageView:
+    def test_not_staff_user_redirected_from_admin_login(self, client, persist_user):
+        response = client.post(reverse('admin:login'), {
+            'username': 'test1',
+            'password': 'PASSWORD',
```
[the remark this answers is not in the document]
Comment (natifridman) [reply tests/main/test_homePage.py:10]: You added it to the test, so you should use it or remove it.
For the password, maybe you can add it as a variable?

[the remark this answers is not in the document]
Comment (shirazyomtov) [reply tests/main/test_homePage.py:10]: For this test, I added the persist_user to create and save the user in the database, so I could check if the user is redirected to the home page after logging in.
So the options are to use persist_user or to create a user and save it in the database, so I would prefers the persist_user option.

If I try to pass it as a variable I still will need to access the user's password(the password is hashed) or I can return it as plain text but in that way there is no advantage to pass it as a variable and I can just use it as plain text in the test itself.
Maybe there is a solution that you know of that can accesses the user's password without it being hashed?

[the remark this answers is not in the document]
Comment (natifridman) [reply tests/main/test_homePage.py:10]: You can't access the password, that's by design.
Why do you need to pass the variable to the test? you can add it to the contest for example, and import it like `from conftest import USERNAME, PASSWORD`, so you won't need to repeat this username/password in other tests.

Please see [this](https://github.com/redhat-beyond/ResuMe/blob/main/tests/users_tests/test_users_views.py) for example

**Review by Omeramsc — APPROVED** [review]
LGTM

File: tests/main/test_homePage.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (shirazyomtov) [reply tests/main/test_homePage.py]: Done

**Review by shacharlevy — DISMISSED** [review]
LGTM!

**Review by natifridman — DISMISSED** [review]
LGTM, please squash the commits

**Review by idan1356 — APPROVED** [review]
LGTM

**Review by shacharlevy — APPROVED** [review]
LGTM!

**Review by natifridman — APPROVED** [review]
